horizon/main.go

```diff
@@ -0,0 +1,113 @@
+package horizon
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/stellar/go-stellar-base/horizon/structs"
+)
+
+const TIMEOUT = 30 * time.Second
+
+// DefaultTestNetClient is a default client to connect to test network
+var DefaultTestNetClient = &Client{URL: "https://horizon-testnet.stellar.org"}
+// DefaultPublicNetClient is a default client to connect to public network
+var DefaultPublicNetClient = &Client{URL: "https://horizon.stellar.org"}
+
+// HorizonError struct contains error and problem returned by Horizon
+type HorizonError struct {
+	Err     error
+	Problem structs.Problem
+}
+
+type HorizonHttpClient interface {
+	Get(url string) (resp *http.Response, err error)
+	PostForm(url string, data url.Values) (resp *http.Response, err error)
+}
+
+// Client struct contains data required to connect to Horizon instance
+type Client struct {
+	URL    string
+	client HorizonHttpClient
+}
+
+// LoadAccount loads the account state from horizon
+func (c *Client) LoadAccount(accountId string) (account structs.Account, horizonError *HorizonError) {
+	httpClient := c.getHttpClient()
+	resp, err := httpClient.Get(c.URL + "/accounts/" + accountId)
+	if err != nil {
+		horizonError = &HorizonError{Err: err}
+		return
+	}
+
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		horizonError = &HorizonError{Err: err}
+		return
+	}
+
+	if resp.StatusCode != 200 {
+		err = errors.New("Error response")
+		horizonError = &HorizonError{Err: err}
+		json.Unmarshal(body, &horizonError.Problem)
+		return
+	}
+
+	err = json.Unmarshal(body, &account)
+	if err != nil {
+		horizonError = &HorizonError{Err: err}
+		return
+	}
+
+	return
+}
+
+// SubmitTransaction submits a transaction to the network
+func (c *Client) SubmitTransaction(transactionEnvelopeXdr string) (response structs.TransactionSuccess, horizonError *HorizonError) {
```

Comment:
Alright, sorry, I think I see where the misunderstanding is.  My two concerns stated plainly.

1.  The function signature of `SubmitTransaction` should use `error`.  I had not seen that you made this change.
2.  HorizonError should not wrap other errors.  This is not fixed.

Let me explain, take the snippet from this PR:

```golang
if err != nil {
 	horizonError = &HorizonError{Err: err}
 	return
}
```

The above code adds no benefit, but actively harms usability because you still have a scenario where